compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -6006,6 +6006,8 @@ public boolean hasFillerValue(BType type) {
                 return tupleType.getTupleTypes().stream().allMatch(eleType -> hasFillerValue(eleType));
             case TypeTags.TYPEREFDESC:
                 return hasFillerValue(getReferredType(type));
+            case TypeTags.INTERSECTION:
+                return hasFillerValue(((BIntersectionType) type).effectiveType);
```

Comment:
Do we have to add this case to the `io.ballerina.runtime.internal.TypeChecker.hasFillerValue(Type, List<Type>)` method also?

Reply:
Reproducible in the following code.
```ballerina
anydata arr = [];
(map<int> & readonly)[2] arr4 = checkpanic arr.cloneWithType();
```
Fixed in the latest commit.